test/upgrade/installation/serverless.go

```diff
@@ -42,5 +49,51 @@ func UpgradeServerless(ctx *test.Context) error {
 		ctx.T.Error("Eventing upgrade failed:", err)
 	}
 
+	// KnativeKafka doesn't provide the version to wait for. Choosing a well-known image and
+	// waiting for all pods to be updated to this image before proceeding.
+	kafkaWebhookImage, err := test.ImageFromCSV(ctx,
+		test.Flags.CSV,
+		test.OperatorsNamespace,
+		"kafka-webhook")
+	if err != nil {
+		ctx.T.Error(err)
+	}
+
+	if err = WaitForPodsWithImage(ctx,
+		knativeEventing,
+		"app=kafka-webhook",
+		"kafka-webhook",
+		kafkaWebhookImage); err != nil {
+		ctx.T.Error(err)
+	}
+
+	if _, err := v1a1test.WaitForKnativeKafkaState(ctx,
+		"knative-kafka",
+		knativeEventing,
+		v1a1test.IsKnativeKafkaReady,
+	); err != nil {
+		ctx.T.Error("KnativeKafka upgrade failed:", err)
+	}
+
+	return nil
+}
+
+func WaitForPodsWithImage(ctx *test.Context, namespace string, podSelector, containerName, expectedImage string) error {
+	if waitErr := wait.PollImmediate(test.Interval, test.Timeout, func() (bool, error) {
+		podList, err := ctx.Clients.Kube.CoreV1().Pods(namespace).List(context.Background(), meta.ListOptions{LabelSelector: podSelector})
+		if err != nil {
+			return false, err
+		}
+		for _, pod := range podList.Items {
+			for _, c := range pod.Spec.Containers {
+				if c.Name == containerName && c.Image != expectedImage {
+					return false, nil
+				}
+			}
+		}
+		return true, nil
```

Comment:
This will be true if the amount of pods is 0. Can that happen?

Also: Do we need to wait for the pods to get ready?

Reply:
Hmm. This is during upgrades so it should never be 0. And if it's 0 then Kafka if not deployed and it can return success, IMO. But I could check if the number of pods is 0 at the beginning.
I don't think we need to wait for the pods to get ready. This should be covered by waiting for KnativeKafka CR.